Add Gree AC example code.

srtulon opened the pull request.

srtulon committed:
add support for Gree AC

crankyoldgit commented on examples/TurnOnGreeAC/TurnOnGreeAC.ino:
```diff
@@ -0,0 +1,77 @@
+*     for your first time. e.g. ESP-12 etc.
+*/
+#include <Arduino.h>
+#endif
```
/home/travis/build/crankyoldgit/IRremoteESP8266/examples/TurnOnGreeAC/TurnOnGreeAC.ino:27:2: error: #endif without #if
 #endif

crankyoldgit commented on examples/TurnOnGreeAC/TurnOnGreeAC.ino:
```diff
@@ -0,0 +1,77 @@
+  ac.setXFan(0);
+  ac.setLight(0);
+  ac.setSleep(0);
+  ac.setTurbo(0);
```
Use `true` instead of `1` and `false` instead of `0`. Those parameters are booleans.

crankyoldgit committed:
Fix some style/linter errors.

e.g.
```
examples/TurnOnGreeAC/TurnOnGreeAC.ino:60:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
examples/TurnOnGreeAC/TurnOnGreeAC.ino:60:  At least two spaces is best between code and comments  [whitespace/comments] [2]
examples/TurnOnGreeAC/TurnOnGreeAC.ino:61:  Line ends in whitespace.  Consider deleting these extra spaces.  [whitespace/end_of_line] [4]
examples/TurnOnGreeAC/TurnOnGreeAC.ino:61:  At least two spaces is best between code and comments  [whitespace/comments] [2]
examples/TurnOnGreeAC/TurnOnGreeAC.ino:62:  Missing space after ,  [whitespace/comma] [3]
```

crankyoldgit commented on examples/TurnOnGreeAC/TurnOnGreeAC.ino:
```diff
@@ -0,0 +1,78 @@
+#include <IRremoteESP8266.h>
+#include <IRsend.h>
+#include <ir_Gree.h>
+#include <ir_Kelvinator.h>
```
Is this line needed?

crankyoldgit commented on examples/TurnOnGreeAC/TurnOnGreeAC.ino:
```diff
@@ -0,0 +1,78 @@
+  Serial.println("Default state of the remote.");
+  printState();
+  Serial.println("Setting desired state for A/C.");
+  ac.off();
```
`ac.off()`?? The filename says it turns the device "On".

srtulon replied:
Hi, I am really sorry about that. I was tinkering with it and somehow didn't save the new version. Thankyou for correcting. I have made the changes

crankyoldgit replied:
Thanks for the fixes

crankyoldgit reviewed (changes requested):
The code you've provided doesn't compile without errors.
Also it doesn't do what the filename suggests (as far as I can see).

Please address the comments and it should be fine to merge.

crankyoldgit committed:
Update copyright info.

also revert the change to `setFan()`

srtulon committed:
Update TurnOnGreeAC.ino

crankyoldgit commented:
FYI, This has been included in the new v2.7.5 release of the library.
